pkg/controllers/repositoryserver/server.go

```diff
@@ -0,0 +1,207 @@
+)
+
+func (h *RepoServerHandler) startRepoProxyServer(ctx context.Context) error {
+	repoServerAddress, err := GetPodAddress(ctx, h.KubeCli, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName)
+	if err != nil {
+		return err
+	}
+	serverAdminUsername := string(h.RepositoryServerSecrets.serverAdmin.Data[serverAdminUserNameKey])
+	serverAdminPassword := string(h.RepositoryServerSecrets.serverAdmin.Data[serverAdminPasswordKey])
+	if err != nil {
+		return errors.Wrap(err, "failed to get pod")
+	}
+	cmd := command.ServerStart(
+		command.ServerStartCommandArgs{
+			CommandArgs: &command.CommandArgs{
+				RepoPassword:   "",
+				ConfigFilePath: command.DefaultConfigFilePath,
+				LogDirectory:   command.DefaultCacheDirectory,
+			},
+			ServerAddress:    repoServerAddress,
+			TLSCertFile:      tlsCertPath,
+			TLSKeyFile:       tlsKeyPath,
+			ServerUsername:   serverAdminUsername,
+			ServerPassword:   serverAdminPassword,
+			AutoGenerateCert: false,
+			Background:       true,
+		},
+	)
+
+	stdout, stderr, err := kube.Exec(h.KubeCli, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, cmd, nil)
+	format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stdout)
+	format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stderr)
+	if err != nil {
+		errors.Wrap(err, "Failed to start Kopia API server")
+	}
+
+	err = h.waitForServerToStart(ctx, repoServerAddress, serverAdminUsername, serverAdminPassword)
+	if err != nil {
+		return errors.Wrap(err, "Failed to start Kopia API server")
+	}
+	err = h.createOrUpdateClientUsers()
+	if err != nil {
+		return errors.Wrap(err, "Failed to create server access users")
+	}
+	return nil
+}
+
+func (h *RepoServerHandler) waitForServerToStart(ctx context.Context, serverAddress, username, password string) error {
+
+	fingerprint, err := kopia.ExtractFingerprintFromCertSecret(ctx, h.KubeCli, h.RepositoryServerSecrets.serverTLS.Name, h.RepositoryServer.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "Failed to extract fingerprint Kopia API Server Certificate Secret Data")
+	}
+	cmd := command.ServerStatus(
+		command.ServerStatusCommandArgs{
+			CommandArgs: &command.CommandArgs{
+				RepoPassword:   "",
+				ConfigFilePath: command.DefaultConfigFilePath,
+				LogDirectory:   command.DefaultLogDirectory,
+			},
+			ServerAddress:  serverAddress,
+			ServerUsername: username,
+			ServerPassword: password,
+			Fingerprint:    fingerprint,
+		})
+
+	ctx, cancel := context.WithTimeout(ctx, DefaultServerStartTimeout)
+	defer cancel()
+
+	err = WaitTillCommandSucceed(ctx, h.KubeCli, cmd, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName)
+	return err
+}
+
+func (h *RepoServerHandler) createOrUpdateClientUsers() error {
+	repoPassword := string(h.RepositoryServerSecrets.repositoryPassword.Data[repoPasswordKey])
```

Comment:
We would not use repo password here. Instead, server control password will be useful.

Reply:
What was the resolution here?

Reply:
```
cmd := command.ServerListUser(
		command.ServerListUserCommmandArgs{
			CommandArgs: &command.CommandArgs{
				RepoPassword:   repoPassword,
				ConfigFilePath: command.DefaultConfigFilePath,
				LogDirectory:   command.DefaultLogDirectory,
			},
		})
		
``` 
Above command needs the repo password, which we are removing in this PR by Ankit - https://github.com/kanisterio/kanister/pull/1778
Once that PR is merged , we can remove this